lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2128,8 +2128,75 @@ TypeSystemSwiftTypeRef::GetByteStride(opaque_compiler_type_t type,
 
 lldb::Encoding TypeSystemSwiftTypeRef::GetEncoding(opaque_compiler_type_t type,
                                                    uint64_t &count) {
-  return m_swift_ast_context->GetEncoding(ReconstructType(type), count);
+  if (!type)
+    return lldb::eEncodingInvalid;
+
+  auto impl = [&]() -> lldb::Encoding {
+    count = 1;
+
+    using namespace swift::Demangle;
+    Demangler dem;
+    auto *node = DemangleCanonicalType(dem, type);
+    auto kind = node->getKind();
+
+    if (kind == Node::Kind::BuiltinTypeName) {
+      assert(node->hasText());
+      if (node->getText().startswith(swift::BUILTIN_TYPE_NAME_INT) ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_WORD)
+        return lldb::eEncodingSint;
+      if (node->getText().startswith(swift::BUILTIN_TYPE_NAME_FLOAT) ||
+          node->getText().startswith(swift::BUILTIN_TYPE_NAME_FLOAT_PPC))
+        return lldb::eEncodingIEEE754;
+      if (node->getText() == swift::BUILTIN_TYPE_NAME_RAWPOINTER ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_NATIVEOBJECT ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_UNSAFEVALUEBUFFER ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_BRIDGEOBJECT)
+        return lldb::eEncodingUint;
+      if (node->getText().startswith(swift::BUILTIN_TYPE_NAME_VEC)) {
+        count = 0;
+        return lldb::eEncodingInvalid;
+      }
+
+      llvm_unreachable("Unhandled builtin");
```

Comment:
Let's do an assert (or an lldb_assert) instead, otherwise this will crash LLDB in production.

Reply:
done